core/src/main/java/me/mykindos/betterpvp/core/command/commands/qol/TeleportPositionCommand.java

```diff
@@ -28,8 +28,38 @@ public String getDescription() {
 
     @Override
     public void execute(Player player, Client client, String... args) {
+        double x = 0, y = 0, z = 0;
         if (args.length == 3) {
-            player.teleport(new Location(player.getWorld(), Double.parseDouble(args[0]), Double.parseDouble(args[1]), Double.parseDouble(args[2])));
+            if (args[0].startsWith("~")) {
+                args[0] = args[0].substring(1);
+                x = player.getX();
+            }
+            try {
+                x += Double.parseDouble(args[0]);
+            } catch (NumberFormatException e) {
+                x += 0;
+            }
+            if (args[1].startsWith("~")) {
+                args[1] = args[1].substring(1);
+                y = player.getY();
+            }
+            try {
+                y += Double.parseDouble(args[1]);
+            } catch (NumberFormatException e) {
+                y +=0;
+            }
+
+            if (args[2].startsWith("~")) {
+                args[2] = args[2].substring(1);
+                z = player.getZ();
+            }
+            try {
+                z += Double.parseDouble(args[2]);
+            } catch (NumberFormatException e) {
+                z += 0;
+            }
```

Comment:
Perhaps some sort of function in UtilFormat or similar would be good here, not required for approval though